runtime-components/non-repudiation-api/src/main/scala/com/daml/nonrepudiation/api/NonRepudiationApi.scala

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.nonrepudiation.api
+
+import java.net.InetAddress
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.server.Directives.{concat, pathPrefix}
+import akka.http.scaladsl.server.Route
+import com.daml.nonrepudiation.api.NonRepudiationApi.Configuration
+import com.daml.nonrepudiation.{CertificateRepository, CommandIdString, SignedPayloadRepository}
+import com.daml.resources.{AbstractResourceOwner, HasExecutionContext, ReleasableResource, Resource}
+
+import scala.concurrent.Future
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+object NonRepudiationApi {
+
+  // We don't need access to the underlying resource, we use
+  // the owner only to manage the server's life cycle
+  def owner[Context: HasExecutionContext](
+      configuration: Configuration,
+      certificateRepository: CertificateRepository,
+      signedPayloadRepository: SignedPayloadRepository.Read[CommandIdString],
+  )(implicit system: ActorSystem): AbstractResourceOwner[Context, Unit] =
+    new NonRepudiationApi[Context](
+      configuration,
+      certificateRepository,
+      signedPayloadRepository,
+    ).map(_ => ())
+
+  final case class Configuration(
+      interface: String,
+      port: Int,
+      shutdownTimeout: FiniteDuration,
+  )
+
+  // Non-repudiation -> NR -> N = 78 and R = 82 in ASCII
```

Comment:
omg, is that how all our weird port numbers are constructed?

Reply:
Wait. I assumed you knew about it.